ingestclient/utils/setup.py

```diff
@@ -0,0 +1,285 @@
+        time_step=time_step, time_step_unit=time_step_unit)
+    try:
+        experiment = rmt.create_project(experiment)
+    except:
+        experiment = rmt.get_project(experiment)
+    return experiment
+
+
+def create_channel(config, rmt, collection, experiment):
+    """
+    Ensure channel exists.
+
+    Args:
+        config (dict):
+        rmt (BossRemote):
+        collection (CollectionResource):
+        experiment (ExperimentResource):
+
+    Returns:
+        (ChannelResource)
+    """
+    if 'sources' not in config['channel']:
+        sources = []
+    else:
+        sources = config['channel']['sources']
+
+    channel = ChannelResource(
+        config['channel']['name'], collection.name, experiment.name,
+        type=config['channel']['type'], description=config['channel']['description'],
+        datatype=config['channel']['datatype'], sources=sources)
+    try:
+        channel = rmt.create_project(channel)
+    except:
+        channel = rmt.get_project(channel)
+    return channel
+
+
+def create_coord_frame(config, rmt):
+    """
+    Ensure coordinate frame exists.
+
+    Args:
+        config (dict):
+        rmt (BossRemote):
+
+    Returns:
+        (CoordinateFrameResource)
+    """
+    coord = CoordinateFrameResource(config['coordinate_frame']['name'],
+                                    config['coordinate_frame']['description'],
+                                    config['coordinate_frame']['x_start'],
+                                    config['coordinate_frame']['x_stop'],
+                                    config['coordinate_frame']['y_start'],
+                                    config['coordinate_frame']['y_stop'],
+                                    config['coordinate_frame']['z_start'],
+                                    config['coordinate_frame']['z_stop'],
+                                    config['coordinate_frame']['voxel_size_x'],
+                                    config['coordinate_frame']['voxel_size_y'],
+                                    config['coordinate_frame']['voxel_size_z'])
+    try:
+        coord = rmt.create_project(coord)
+    except:
+        coord = rmt.get_project(coord)
+    return coord
+
+
+def main(args):
+    """
+    Main entry point of script.
+
+    Args:
+        args (Namespace): Parsed command line arguments.
+    """
+    rmt = BossRemote(args.intern_cfg)
+
+    # Load Configuration File
+    with open(args.config_file, 'rt') as cfg:
+        config = json.load(cfg)
+
+    collection = create_collection(config, rmt)
+    coord = create_coord_frame(config, rmt)
+    experiment = create_experiment(config, rmt, collection, coord)
+    channel = create_channel(config, rmt, collection, experiment)
+
+    # Add gov team permissions
+    try:
+        rmt.add_permissions(GOV_TEAM, collection, ['read'])
+        rmt.add_permissions(GOV_TEAM, experiment, ['read'])
+        rmt.add_permissions(GOV_TEAM, channel, ['read'])
+    except:
```

Comment:
Should the try/except be around each permissions? Right now the user won't know which permissions failed (or why). Also thinking about one `add_permissions` failing but another working...

Reply:
Sure, that'd be helpful.